bon-macros/src/builder/builder_gen/input_func.rs

```diff
@@ -503,3 +504,27 @@ impl Visit<'_> for FindSelfReference {
         }
     }
 }
+
+fn get_must_use_attribute(attrs: &[syn::Attribute]) -> Option<syn::Attribute> {
+    let mut iter = attrs
+        .iter()
+        .filter(|attr| !attr.is_doc() && matches!(attr.style, syn::AttrStyle::Outer))
+        .filter(|attr| {
+            let path = match &attr.meta {
+                syn::Meta::Path(path) => path,
+                syn::Meta::List(_) => return false,
+                syn::Meta::NameValue(name_value) => &name_value.path,
+            };
+            path.segments.len() == 1
+                && path
+                    .segments
+                    .iter()
+                    .any(|segment| segment.ident.raw_name() == "must_use")
+        });
+    let result = iter.next();
+    if iter.next().is_some() {
+        // Fishy: multiple must_use? panic here?
```

Comment:
This results in a compiler warning:
```rust
#[must_use = "first"]
#[must_use = "second"]
fn foo() {}
```
```
warning: unused attribute
 --> crates/sandbox/src/lib.rs:2:1
  |
2 | #[must_use = "second"]
  | ^^^^^^^^^^^^^^^^^^^^^^ help: remove this attribute
  |
note: attribute also specified here
 --> crates/sandbox/src/lib.rs:1:1
  |
1 | #[must_use = "first"]
  | ^^^^^^^^^^^^^^^^^^^^^
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: `#[warn(unused_attributes)]` on by default
```
I think in this case we should just return an error, with the span pointing to the second usage of the attribute e.g.:
```rust
if let Some(second_attr) = iter.next() {
    bail!(second_attr, "Found multiple #[must_use], but only one is allowed");    
}
```
This way the error message will point to the second `#[must_use]` attribute. You can test this by adding a test case to `/home/veetaha/work/bon/bon/tests/integration/ui/compile_fail/misc.rs`. Then update the `misc.stderr` snapshot with 
```
TRYBUILD=overwrite cargo t --all-features --all-targets -p bon
```
And inspect the git diff in the error messages in that stderr output.

Btw. I also recommend you to add a test that creates a struct and doesn't use it or calls a function with `must_use` without using its result into the `compile_fail/misc.rs` file. This file is used to test that the compiler is able to generate the expceted error messages and warnings.